File: c99/bc1094.c
#include <stdio.h>
#include <string.h>
#include <conio.h>

int main()
{
	int i, qTotal, qCo, qSa, qRa, iVec, fVal;
	qTotal = qCo = qSa = qRa = 0;
	char li[3];
	char tAn;
	
	scanf("%d", &iVec);
	//fflsuh(stdin);
	
	for(i = 0; i < iVec; i++)
	{
		fgets(li);
		
		char val = li[2];
		
		fVal = (int) li[0];
		strcpy(tAn, val);
		
		_Bool fVer = strcmp(tAn, "C");
		_Bool sVer = strcmp(tAn, "R");
		_Bool tVer = strcmp(tAn, "S");
		
		qTotal += fVal;
		
		if(fVer == 0) qCo += fVal;
		else if(sVer == 0) qRa += fVal;
		else if(tVer == 0) qSa += fVal;
		
	}
	
	printf("Total: %d cobaias\n", qTotal);
	printf("Total de coelhos: %d\nTotal de ratos: %d\nTotal de sapos: %d\n", qCo, qRa, qSa);
	printf("Percentual de coelhos: %.2f %\n", (double) qCo / qTotal*100);
	printf("Percentual de ratos: %.2f %\n", (double) qRa / qTotal*100);
	printf("Percentual de sapos: %.2f %\n", (double) qSa / qTotal*100);
	
}

File: python/bc1094.py
qtdE = int(input())
tCo = 0
tRa = 0
tSa = 0
total = 0

for c in range(1, qtdE+1):
    cVal = input().split()
    qtd = int(cVal[0])
    tC = cVal[1]

    if tC == 'C': tCo += qtd
    elif tC == 'R': tRa += qtd
    elif tC == 'S': tSa += qtd

    total += qtd

print(f"Total: {total} cobaias\nTotal de coelhos: {tCo}\nTotal de ratos: {tRa}\nTotal de sapos: {tSa}")
print(f"Percentual de coelhos: {tCo/total*100:.2f} %\nPercentual de ratos: {tRa/total*100:.2f} %\nPercentual de sapos: {tSa/total*100:.2f} %")
